app/lib/forms/choose_your_provider.rb

```diff
@@ -22,10 +22,12 @@ def previous_step
         :funding_your_npq
       elsif course.npqh? && eligible_for_funding?
         :possible_funding
-      elsif course.aso? && wizard.store["aso_funding"] == "yes"
-        :funding_your_aso
-      elsif wizard.store["aso_new_headteacher"] == "yes"
-        :aso_possible_funding
+      elsif course.ehco?
+        if eligible_for_funding?
+          :aso_possible_funding
+        else
+          :funding_your_aso
+        end
```

Comment:
This section was duplicating eligibility funding logic to determine whether the user came from the eligible or ineligible path, if we leverage eligible_for_funding we can remoev this duplication and improve accuracy of this calculation.